app/src/testWebkit/java/org/mozilla/focus/telemetry/FirebaseEventTest.java

```diff
@@ -0,0 +1,138 @@
+
+    final private static int EVENT_NAME_SEPARATOR_LENGTH = FirebaseEvent.EVENT_NAME_SEPARATOR.length() * 2;
+
+    static {
+        // event name already contains two "__", so they are 36 characters left
+        INVALID_EVENT_NAME = prepareString(FirebaseEvent.MAX_LENGTH_EVENT_NAME - EVENT_NAME_SEPARATOR_LENGTH + 1);
+        VALID_EVENT_NAME = prepareString(FirebaseEvent.MAX_LENGTH_EVENT_NAME - EVENT_NAME_SEPARATOR_LENGTH);
+
+        INVALID_PARAM_NAME = prepareString(FirebaseEvent.MAX_LENGTH_PARAM_NAME + 1);
+        VALID_PARAM_NAME = prepareString(FirebaseEvent.MAX_LENGTH_PARAM_NAME);
+
+        INVALID_PARAM_VALUE = prepareString(FirebaseEvent.MAX_LENGTH_PARAM_VALUE + 1);
+        VALID_PARAM_VALUE = prepareString(FirebaseEvent.MAX_LENGTH_PARAM_VALUE);
+
+    }
+
+    private static String prepareString(final int size) {
+        char[] chars = new char[size];
+        Arrays.fill(chars, FAKE_EVENT_NAME_CHAR);
+        return new String(chars);
+    }
+
+    @NonNull
+    private FirebaseEvent generateValidFirebaseEvent() {
+        return FirebaseEvent.create("", "", "", VALID_EVENT_NAME);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void eventNameLongerThan40Characters_shouldThrowIllegalArgumentException() {
+        FirebaseEvent.create("", "", "", INVALID_EVENT_NAME);
+    }
+
+    @Test
+    public void eventNameValid_shouldBeSafe() {
+        try {
+            generateValidFirebaseEvent();
+        } catch (Exception e) {
+            fail("Should not have thrown any exception");
+        }
+    }
+
+
+    @Test
+    public void nullContext_shouldBeSafe() {
+        try {
+            generateValidFirebaseEvent().queue(null);
+        } catch (Exception e) {
+            fail("Should not have thrown any exception");
+        }
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void paramSizeTooLarge_shouldThrow() {
+        final FirebaseEvent firebaseEvent = generateValidFirebaseEvent();
+        final Bundle params = mock(Bundle.class);
+        firebaseEvent.setParam(params);
+        when(params.size()).thenReturn(FirebaseEvent.MAX_PARAM_SIZE + 1);
+        firebaseEvent.param("", "");
+    }
+
+    @Test
+    public void paramSizeValid_shouldBeSafe() {
+        final FirebaseEvent firebaseEvent = generateValidFirebaseEvent();
+        final Bundle extras = mock(Bundle.class);
+        firebaseEvent.setParam(extras);
+
+        when(extras.size()).thenReturn(FirebaseEvent.MAX_PARAM_SIZE);
+        try {
+            firebaseEvent.param("", "");
+        } catch (Exception e) {
+            fail("Should not have thrown any exception");
+        }
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void paramKeyLengthTooLarge_shouldThrow() {
+        final FirebaseEvent firebaseEvent = generateValidFirebaseEvent();
+        firebaseEvent.param(INVALID_PARAM_NAME, "");
+    }
+
+    @Test
+    public void paramKeyValid_shouldBeSafe() {
```

Comment:
I think this test is somewhat duplicating paramKeyLengthTooLarge_shouldThrow. Other pairs are duplications as well IMO.

Reply:
fixed